Add XL_PATH to allow alternate config directories

rankynbass opened the pull request:
XL_PATH can now be used to set an alternate config directory. This is useful for multiboxing, and for people who want their config in `~/.local/share/xlcore` instead of `~/.xlcore`.

Also updated the README with a few environment variables.

rankynbass committed:
Add XL_PATH to allow alternate config directories

goaaats commented:
Could this be XLCORE_PATH? Might make it clear that this only works on xlcore. Or we also implement it on Windows and name it XL_USERDIR or something.

rankynbass committed:
Change XL_PATH to XL_USERDIR
* update readme to match
* remove -RB from readme

rankynbass committed:
Remove XL_FIRSTRUN from readme

rankynbass commented:
Alright, I've removed XL_FIRSTRUN from the readme. I'm hesitant to remove the implimentation because it's still checked in a couple of places, and I don't want to break anything.

I have also fixed the readme to no longer reference -RB, and changed XL_PATH to XL_USERDIR.

acfrazier commented:
@reiichi001 Is this able to be merged? I was trying to migrate my multibox setup to Linux this morning, and realized that neither XL_PATH nor XL_USERDIR are respected.

Blooym committed:
Merge branch 'main' into add-xl-path
